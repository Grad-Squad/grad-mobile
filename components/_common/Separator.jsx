import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import { Colors } from 'styles';

const Separator = () => <View style={styles.separator} />;

Separator.propTypes = {};
Separator.defaultProps = {};

export default React.memo(Separator);

const styles = StyleSheet.create({
  separator: {
    width: '100%',
    marginVertical: 10,
    height: 1.2,
    backgroundColor: Colors.separator,
  },
});
