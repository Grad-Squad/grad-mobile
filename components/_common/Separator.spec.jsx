import { render } from '@testing-library/react-native';
import React from 'react';
import Separator from './Separator';

it('should match snapshot', () => {
  const separator = render(<Separator />);
  expect(separator.toJSON()).toMatchSnapshot();
});

it('should match snapshot with style prop', () => {
  const separator = render(<Separator style={{ backgroundColor: 'pink' }} />);
  expect(separator.toJSON()).toMatchSnapshot();
});
