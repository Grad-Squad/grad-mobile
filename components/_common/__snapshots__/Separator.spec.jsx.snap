// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dddddd",
        "height": 1.2,
        "marginVertical": 10,
        "width": "100%",
      },
      Object {},
    ]
  }
/>
`;

exports[`should match snapshot with style prop 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dddddd",
        "height": 1.2,
        "marginVertical": 10,
        "width": "100%",
      },
      Object {
        "backgroundColor": "pink",
      },
    ]
  }
/>
`;
